authority/ssh.go

```diff
@@ -241,6 +242,45 @@ func (a *Authority) SignSSH(ctx context.Context, key ssh.PublicKey, opts provisi
 		return nil, errs.InternalServer("authority.SignSSH: unexpected ssh certificate type: %d", certTpl.CertType)
 	}
 
+	switch certTpl.CertType {
+	case ssh.UserCert:
+		// when no user policy engine is configured, but a host policy engine is
+		// configured, the user certificate is denied.
+		if a.sshUserPolicy == nil && a.sshHostPolicy != nil {
+			return nil, errs.ForbiddenErr(errors.New("authority not allowed to sign ssh user certificates"), "authority.SignSSH: error creating ssh user certificate")
+		}
```

Comment:
I don't understand why do we need or want this condition.

Reply:
The goal is to not allow a SSH user cert when an SSH host policy has been configured. The reasoning for this is that users creating a policy for SSH host certificates intend to do so just for host certificates. When they don't specify a SSH user policy, all SSH user certificates will not be allowed. A user should create both an SSH host and user policy if both types of certs are allowed. The logic is performed like this, because the policy engines are not aware of each other.